bio_hansel/subtyper.py

```diff
@@ -228,6 +228,30 @@ def parallel_query_reads(reads: List[Tuple[List[str], str]],
     outputs = [x.get() for x in res]
     return outputs
 
+def calc_kmer_fraction(df,min_kmer_frac=0.05):
+    """Calculate the percentage each k-mer frequence represents for a given position
+
+    Args:
+        df: BioHansel k-mer frequence pandas df
+        min_kmer_frac: float 0 - 1 on the minimum fraction a kmer needs to be to be considered valid
+
+    Returns:
+        - pd.DataFrame with k-mers with kmer_fraction column
+    """
+    position_frequencies = df[['refposition','freq']].groupby(['refposition']).sum().reset_index()
```

Comment:
It might be easier to use a dict:

```python
position_frequencies = df[['refposition','freq']].groupby(['refposition']).sum().to_dict()
```

The dict should have `refposition` keys and summed frequency values, so getting `total_freq` would be easier and clearer:

```python
total_freq = position_frequencies[row.refposition]
```